gnmi/gnmi.go

```diff
@@ -112,28 +191,147 @@ func (g *GNMI) handleMetrics(_ gnmit.Queue, updateFn gnmit.UpdateFn, target stri
 	go func() {
 		for ms := range g.metricCh {
 			var updates []*gpb.Update
+
+			// Iterate over all resources (e.g., app).
 			rms := ms.ResourceMetrics()
 			for i := 0; i < rms.Len(); i++ {
 				rm := rms.At(i)
+
+				// Iterate over all instrument scopes within the resource (e.g., module within an app).
 				ilms := rm.ScopeMetrics()
 				for j := 0; j < ilms.Len(); j++ {
 					ilm := ilms.At(j)
+
+					// Iterate over all metrics for the instrument scope.
 					ms := ilm.Metrics()
 					for k := 0; k < ms.Len(); k++ {
 						m := ms.At(k)
+
+						// Handle each metric type.
 						switch m.Type() {
+						case pmetric.MetricTypeEmpty:
+							g.logger.Error("empty metric type", zap.String("name", m.Name()))
+							continue
+						case pmetric.MetricTypeGauge:
+							gaugeMetrics := m.Gauge().DataPoints()
+							for l := 0; l < gaugeMetrics.Len(); l++ {
+								gaugeMetric := gaugeMetrics.At(l)
+
+								// Data can be integers or doubles.
```

Comment:
Could this switch be broken out into a function?

Reply:
done.